agent/task.go

```diff
@@ -228,8 +229,23 @@ func (a *Agent) getTaskConfig(ctx context.Context, tc *taskContext) (*model.Task
 		return nil, errors.New("agent retrieved an empty project ref")
 	}
 
+	var confPatch *patch.Patch
+	if confVersion.Requester == evergreen.GithubPRRequester {
+		tc.logger.Execution().Info("Fetching patch.")
+		confPatch, err = a.comm.GetTaskPatch(ctx, tc.task)
+		if err != nil {
+			return nil, err
+		}
+		if confPatch == nil {
+			return nil, errors.New("agent retrieved an empty patch")
```

Comment:
what would cause confPatch to be nil and err to not be nil? will we ever hit this case?

Reply:
I changed the client method to generate an error when the API doesn't respond 200 OK, so it's now impossible. I've also removed this code.